code/modules/clothing/modular/attachables.dm

```diff
@@ -0,0 +1,118 @@
+/**
+	Modular armor attachments
+
+	These are utilty attachments that equip into module slots on modular armor
+*/
+
+/** Shoulder lamp stength module */
+/obj/item/armor_module/attachable/better_shoulder_lamp
+	name = "Baldur Light Amplification System"
+	desc = "Designed for mounting on the Jaeger Combat Exoskeleton. Substantially increases the power output of the Jaeger Combat Exoskeletons mounted flashlight. Doesn’t really slow you down."
+	icon_state = "mod_lamp_icon"
+	item_state = "mod_lamp"
+	var/power_boost = 4 /// The boost to armor shoulder light
+
+/obj/item/armor_module/attachable/better_shoulder_lamp/do_attach(mob/living/user, obj/item/clothing/suit/modular/parent)
+	. = ..()
+	parent.light_strength += power_boost
+	parent.update_overlays()
+
+/obj/item/armor_module/attachable/better_shoulder_lamp/do_detach(mob/living/user, obj/item/clothing/suit/modular/parent)
+	parent.light_strength -= power_boost
+	return ..()
+
+
+/** Shoulder lamp stength module */
+/obj/item/armor_module/attachable/valkyrie_autodoc
+	name = "Valkyrie Automedical Armor System"
+	desc = "Designed for mounting on the Jaeger Combat Exoskeleton. Has a variety of chemicals it can inject, as well as automatically securing the bones and body of the wearer, to minimise the impact of broken bones or mangled limbs in the field. Will definitely impact mobility."
+	icon_state = "mod_autodoc_icon"
+	item_state = "mod_autodoc"
+	slowdown = 0.25
+
+/obj/item/armor_module/attachable/valkyrie_autodoc/do_attach(mob/living/user, obj/item/clothing/suit/modular/parent)
+	. = ..()
+	var/list/tricord = list(/datum/reagent/medicine/tricordrazine)
+	var/list/tramadol = list(/datum/reagent/medicine/tramadol)
+	parent?.AddComponent(/datum/component/suit_autodoc, 2.5 MINUTES, tricord, tricord, tricord, tricord, tramadol, 0.5)
```

Comment:
```suggestion
	parent.AddComponent(/datum/component/suit_autodoc, 2.5 MINUTES, tricord, tricord, tricord, tricord, tramadol, 0.5)
```
Isn't it guaranteed that there will be a parent here?

Reply:
This was a time when you could have two autodocs modules and you may break things having these weird interactions, but now it should be defined as you stated.